app/src/main/res/layout/product_popup.xml

```diff
@@ -134,4 +135,12 @@
 
     </LinearLayout>
 
+    <TextView
+        android:id="@+id/lblProductUpc"
```

Comment:
This is the hidden text view for the 'product.upc'.
Although it does not matter much, it is probably supposed to be something else like a label or simple text component.